pac4j-oidc/src/main/java/org/pac4j/oidc/client/OidcClient.java

```diff
@@ -294,9 +297,23 @@ protected void internalInit(final WebContext context) {
             throw new TechnicalException(e);
         }
 
-        final ClientAuthenticationMethod method = getProviderMetadata().getTokenEndpointAuthMethods() != null
-                && getProviderMetadata().getTokenEndpointAuthMethods().size() > 0 ? getProviderMetadata()
-                .getTokenEndpointAuthMethods().get(0) : ClientAuthenticationMethod.getDefault();
+        // check authentication methods
+        final List<ClientAuthenticationMethod> methods = getProviderMetadata().getTokenEndpointAuthMethods();
+        final ClientAuthenticationMethod method;
+
+        if (methods != null && methods.size() > 0 ) {
+            if (methods.contains(getClientAuthenticationMethod())) {
+                method = getClientAuthenticationMethod();
+            } else {
+                method = getProviderMetadata().getTokenEndpointAuthMethods().get(0);
+                logger.warn("Preferred token endpoint Authentication method: {} not available. Defaulting to: {}",
+                        getClientAuthenticationMethod(), method);
+            }
+        } else {
+            method = ClientAuthenticationMethod.getDefault();
+            logger.warn("Provider metadata does not provide Token endpoint authentication methods. Defaulting to: {}",
```

Comment:
What does this mean for a client to not have a token endpoint? How do you get tokens?